app/src/org/commcare/activities/CommCareSetupActivity.java

```diff
@@ -522,10 +522,12 @@ public boolean onPrepareOptionsMenu(Menu menu) {
         }
         
         MenuItem refreshItem = menu.findItem(MENU_REFRESH_OPPORTUNITIES);
-        boolean showRefreshMenu =  !fromExternal &&
-                PersonalIdManager.getInstance().isloggedIn() &&
-                !ConnectUserDatabaseUtil.hasConnectAccess(this);
-        refreshItem.setVisible(showRefreshMenu);
+        if (refreshItem != null) {

```

Comment:
Any idea what's causing this to be `null` ?

Reply:
I tried to replicate a few times but I wasn't able to. What I noticed is that we call `invalidateOptionsMenu()` in a few places and that when we start the activity, the method `onPrepareOptionsMenu()` is called twice. The latter makes me think that maybe when the activity is recreated (due to a screen rotation), a delay is introduced, I tried this and couldn't replicate still.